ps_democqrshooksusage.php

```diff
@@ -0,0 +1,337 @@
+        $searchCriteria = $params['search_criteria'];
+
+        $searchQueryBuilder->addSelect(
+            'IF(dcur.`is_allowed_for_review` IS NULL,0,dcur.`is_allowed_for_review`) AS `is_allowed_for_review`'
+        );
+
+        $searchQueryBuilder->leftJoin(
+            'c',
+            '`' . pSQL(_DB_PREFIX_) . 'democqrshooksusage_reviewer`',
+            'dcur',
+            'dcur.`id_customer` = c.`id_customer`'
+        );
+
+        if ('is_allowed_for_review' === $searchCriteria->getOrderBy()) {
+            $searchQueryBuilder->orderBy('dcur.`is_allowed_for_review`', $searchCriteria->getOrderWay());
+        }
+
+        foreach ($searchCriteria->getFilters() as $filterName => $filterValue) {
+            if ('is_allowed_for_review' === $filterName) {
+                $searchQueryBuilder->andWhere('dcur.`is_allowed_for_review` = :is_allowed_for_review');
+                $searchQueryBuilder->setParameter('is_allowed_for_review', $filterValue);
+
+                if (!$filterValue) {
+                    $searchQueryBuilder->orWhere('dcur.`is_allowed_for_review` IS NULL');
+                }
+            }
+        }
+    }
+
+    /**
+     * Hook allows to modify Customers form and add aditional form fields as well as modify or add new data to the forms.
+     *
+     * @param array $params
+     */
+    public function hookactioncustomerFormBuilderModifier(array $params)
+    {
+        /** @var FormBuilderInterface $formBuilder */
+        $formBuilder = $params['form_builder'];
+        $formBuilder->add('is_allowed_for_review', SwitchType::class);
+
+        /** @var CommandBusInterface $queryBus */
+        $queryBus = $this->get('prestashop.core.query_bus');
+
+        /** @var ReviewerSettingsForForm $reviewerSettings */
+        $reviewerSettings = $queryBus->handle(new GetReviewerSettingsForForm($params['id']));
+
+        $params['data']['is_allowed_for_review'] = $reviewerSettings->isAllowedForReview();
+
+        $formBuilder->setData($params['data']);
+    }
+
+    /**
+     * Hook allows to modify Customers form and add aditional form fields as well as modify or add new data to the forms.
+     *
+     * @param array $params
+     *
+     * @throws CustomerException
+     */
+    public function hookactionAfterUpdatecustomerFormHandler(array $params)
+    {
+        $this->updateCustomerReviewStatus($params);
+    }
+
+    /**
+     * Hook allows to modify Customers form and add aditional form fields as well as modify or add new data to the forms.
+     *
+     * @param array $params
+     *
+     * @throws CustomerException
+     */
+    public function hookactionAfterCreatecustomerFormHandler(array $params)
+    {
+        $this->updateCustomerReviewStatus($params);
+    }
+
+    /**
+     * @param array $params
+     *
+     * @throws CustomerException
+     */
+    private function updateCustomerReviewStatus(array $params)
+    {
+        //todo: a better would be to grab the data from array?
```

Comment:
Can you explain ? 🤔 I dont get it

Reply:
I had here one idea but recently decided that it not suitable -  never mind here, removed the todo :D